### DISCUSSION TITLE
Nuxt3.0.0 NuxtLayout not receiving custom props.

### DISCUSSION THREAD

**Discussion by ateliee** [nuxt/nuxt#16180] [Questions]
Hello,

NuxtLayout received custom props in Nuxt3.0.0 rc-12.
vue shows a warning when upgrading to Nuxt3.0.0.

Is there any solution?

```
// layouts/default.vue
<template>
    <div>
       {{ props.text }}
       <slot />
    </div>
</template>

<script lang="ts" setup>
const props = defineProps<{
    text: string,
}>();
</script>
```

```
// pages/index.vue
<template>
    <NuxtLayout text="test">
       some
    </NuxtLayout>
</template>
```

console output is

```
[Vue warn]: Mission required prop: "text"
```

**Comment by jamiemcinnes** [comment]
Agreed. It did on rc-12 but I've just updated and now NuxtLayout doesnt seem to support custom props. I get "Extraneous non-props attributes were passed to component but could not be automatically inherited because component renders fragment or text root nodes."

**Comment by TheAlexLichter** [comment]
Should work again via https://github.com/nuxt/framework/pull/9395

**Reply by bogdan0083** [reply]
@manniL Hi! I'm sorry for pinging you. Is it possible to install latest nuxt version directly from github before it gets released? I'm having the same issue, I see that the PR is already landed in `main` branch but it's not available under `npm install nuxt`. 

Thanks! :heart:

**Reply by TheAlexLichter** [reply]
Yes, try out the [edge channel](https://v3.nuxtjs.org/guide/going-further/edge-channel/) ☺️

**Reply by bogdan0083** [reply]
> Yes, try out the [edge channel](https://v3.nuxtjs.org/guide/going-further/edge-channel/) ☺️

Wow, didn't know about this. Thanks again!

**Reply by TheAlexLichter** [reply]
No problem 🙌🏻
PS: Feel free to mark the reply as answer 👍🏻
